replay/models/base_rec.py

```diff
@@ -1694,14 +1694,17 @@ def _predict_without_sampling(
             ),
         )
 
-        max_hist_len = (
-            self._calc_max_hist_len(log, users)
-            if filter_seen_items and log is not None
-            else 0
-        )
+        if filter_seen_items and log is not None:
+            user_to_num_items = log.groupBy("user_idx").agg(
+                sf.countDistinct("item_idx").alias("num_items")
+            )
+            users = users.join(user_to_num_items, on="user_idx")
```

Comment:
what will you get if a user is absent is `log`. seems you will lose the user in this join. it should be right join